---
title: "Draft paper"
author: "Nick Sunderland"
date: "2024-12-04"
output: html_document
bibliography: "`r BibTexLibraries::get_bib_path('dcm_pgs')`"
csl: "`r BibTexLibraries::get_bib_format('american-medical-association-no-et-al.csl')`"
link-citations: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
dir <- here::here()
library(data.table)
library(ggplot2)
library(scales)
library(plotly)
library(reticulate)
reticulate::py_run_string("import plotly")
reticulate::py_run_string("import kaleido")
library(viridis)
library(splines)
library(kableExtra)
```

## Set default parameters
```{r}
dcm_prev      <- 0.004 # DCM prevalence
gene_pos_pct  <- 0.35  # proportion of DCM cases gene positive
gene_vus_pct  <- 0.15  # proportion of DCM cases variant of unknown significance
gene_neg_pct  <- 0.50  # proportion of DCM cases gene negative
n_individuals <- 500000
```

## Simulate data
Here we simulate a dataset of `r format(n_individuals, big.mark = ",", scientific=FALSE)` individuals using a DCM prevalence of `r sprintf("%0.2f%%", dcm_prev*100)` and percentage of gene positive, gene negative, and variant of unknown significance (VUS) of `r sprintf("%.0f%%", gene_pos_pct*100)`, `r sprintf("%.0f%%", gene_neg_pct*100)`, and `r sprintf("%.0f%%", gene_vus_pct*100)`, respectively. We adjust the means of the polygenic risk score such that the gene positive cohort has the maximum, followed by VUS, then gene negative. Non-DCM individuals have a mean PGS of 0.

```{r, warning=FALSE, message=FALSE}
# set prevalences and PGS means
gene_statuses <- list(
  `No DCM`       = list(prop = 1-dcm_prev,            mean_pgs = 0.00),  
  `DCM gene +ve` = list(prop = dcm_prev*gene_pos_pct, mean_pgs = 1.00), 
  `DCM gene vus` = list(prop = dcm_prev*gene_vus_pct, mean_pgs = 0.85), 
  `DCM gene -ve` = list(prop = dcm_prev*gene_neg_pct, mean_pgs = 0.60)   
)

# reproducibility
set.seed(123)

# simulate te dataset
dat <- data.table(
  id          = sprintf(paste0("ID_%0", ceiling(log10(n_individuals+1)), "d"), 1:n_individuals),
  sex         = factor(sample(c("male", "female"), n_individuals, replace = TRUE), levels = c("male", "female")),
  gene_status = factor(sample(x       = names(gene_statuses), 
                              size    = n_individuals, 
                              replace = TRUE, 
                              prob    = sapply(gene_statuses, function(x) x$prop)), 
                      levels = names(gene_statuses))
)
dat[, dcm_status := factor(gene_status=="No DCM", levels=c(T,F), labels=c("No DCM", "DCM"))]
dat[, pgs := rnorm(.N, mean = gene_statuses[[as.character(gene_status)]]$mean_pgs, sd = 1), by = .(gene_status)]
dat[, age := ifelse(dcm_status == "DCM", 
                    pmin(pmax(rnorm(.N, mean = 55, sd = 10), 18), 95), 
                    pmin(pmax(rnorm(.N, mean = 45, sd = 15), 18), 95))] 
dat[, age_cat := cut(age,
                     breaks = seq(0, 100, by = 20),
                     labels = c("0-19", "20-39", "40-59", "60-79", "80+"), 
                     right = FALSE)]

# view
knitr::kable(unique(dat, by="gene_status"))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.show="hold", fig.align='center', out.width="50%"}
# plot pie chart
p <- ggplot(data = (dat[gene_status != "No DCM", .N, by = gene_status
                       ][
                        , prop := N / sum(N)
                       ]), 
       aes(x = "", y = prop, fill = gene_status)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  scale_fill_viridis_d() +
  labs(fill = "Gene status") +
  theme_void(base_size = 22)
p
png(file.path(dir, "figures", "dcm_proportions.png"), width=600, height=600)
p
invisible(dev.off())

# plot PGS distributions
p1 <- ggplot(dat, aes(x = pgs, fill = gene_status)) +
  geom_density(alpha=0.3) +
  scale_fill_viridis_d() +
  labs(x = "PGS", fill = "Gene status") +
  theme_classic(base_size = 22) +
  theme(axis.text.x  = element_blank(), 
        axis.ticks.x = element_blank(), 
        axis.text.y  = element_blank(), 
        axis.ticks.y = element_blank(), 
        axis.title.y = element_blank())
p1
png(file.path(dir, "figures", "psg_distributions.png"), width=800, height=600)
p1
invisible(dev.off())
```

---

## Odds ratio by DCM PGS quantile
We then calculate the odds ratio for DCM by PGS bin. We define appropriate cut-offs for the PGS for "pathogenic" and "intermediate" criteria.

### PGS odds ratio criteria
```{r, echo=F}
pgs_thresholds <- data.table(
  or_thresh  = c(1.5, 5.0), 
  thresh_lab = c("intermediate", "pathogenic") 
)
knitr::kable(pgs_thresholds) |> 
  kableExtra::kable_styling(bootstrap_options = c("striped", "hover"), full_width = FALSE)
```

### Odds ratio 
We calculate the odds ratio by PGS bin and fit a spline model to help extract the quantile at which the thresholds are reached.

```{r, warning=FALSE, message=FALSE}
# calculate the odds ratio of DCM by PGS quantile 
n_quantiles <- 10
quantiles   <- c(0, 1, 5, 10, 20, 40, 60, 80, 90, 95, 99, 100)
dat[, pgs_quantile := cut(pgs, 
                          breaks = quantile(pgs, probs = quantiles/100, na.rm = TRUE), 
                          labels = c(paste0("<",quantiles[2]), 
                                     paste0(quantiles[2:(length(quantiles)-2)], "-", quantiles[3:(length(quantiles)-1)]), 
                                     paste0(">", quantiles[length(quantiles)-1])), 
                          include.lowest = TRUE)]
dat[, pgs_percentile := cut(pgs, 
                            breaks = quantile(pgs, probs = seq(0,1,0.01), na.rm = TRUE), 
                            labels = seq(1,100,1), 
                            include.lowest = TRUE)]
or_table <- data.table(pgs_quantile = unique(dat$pgs_quantile))
or_table[, `:=`(
  dcm_cases_in_group    = dat[pgs_quantile == .BY[[1]] & dcm_status != "No DCM", .N],
  dcm_controls_in_group = dat[pgs_quantile == .BY[[1]] & dcm_status == "No DCM", .N],
  dcm_cases_rest        = dat[pgs_quantile != .BY[[1]] & dcm_status != "No DCM", .N],
  dcm_controls_rest     = dat[pgs_quantile != .BY[[1]] & dcm_status == "No DCM", .N]
), by = pgs_quantile]
or_table[, `:=`(
  odds_in_group = dcm_cases_in_group / dcm_controls_in_group,
  odds_in_rest  = dcm_cases_rest / dcm_controls_rest
)]
or_table[, `:=`(odds_ratio    = odds_in_group / odds_in_rest, 
                se_log_or     = sqrt(1 / dcm_cases_in_group + 
                                     1 / dcm_controls_in_group + 
                                     1 / dcm_cases_rest + 
                                     1 / dcm_controls_rest))]
or_table[, `:=`(log_or   = log(odds_ratio))]
or_table[, `:=`(ci_lower = exp(log_or - 1.96 * se_log_or),
                ci_upper = exp(log_or + 1.96 * se_log_or))]

# model the odds ratio across PGS quantile and extract the thresholds of 5 and 1.5
or_table[, pgs_midpoint := as.numeric(sub("-.*", "", as.character(pgs_quantile))) + (as.numeric(sub(".*-", "", as.character(pgs_quantile))) - as.numeric(sub("-.*", "", as.character(pgs_quantile)))) / 2]
or_table[as.character(pgs_quantile) == "<1", pgs_midpoint := 0.5]
or_table[as.character(pgs_quantile) == ">99", pgs_midpoint := 99.5]

spline_model <- lm(log(odds_ratio) ~ ns(pgs_midpoint, df = 6), data = or_table)
pred_pgs_values <- seq(0,100,0.01)
predicted_or <- data.table(pgs_quantile_val = pred_pgs_values, 
                           pred_or      = exp(predict(spline_model, newdata = data.frame(pgs_midpoint = pred_pgs_values))))

# predict the thresholds from the fit 
pgs_thresholds[, pgs_cutoff := sapply(or_thresh, function(x) predicted_or[which.min(abs(pred_or-x)), pgs_quantile_val])]
pgs_thresholds[, pgs := sapply(pgs_cutoff, function(x) dat[ceiling(x)==as.numeric(pgs_percentile), mean(pgs, na.rm=T)])]

dat <- dat[
  pgs_thresholds, 
  on = .(pgs > pgs), 
  `:=`(thresh_lab = paste0(i.thresh_lab, " PGS"))
]
dat[is.na(thresh_lab), `:=`(thresh_lab = "low PGS")]

# view
knitr::kable(unique(dat, by="thresh_lab"))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.align='center', out.width="49%"}
# plot OR of DCM by PGS quantile 
p3 <- ggplot(or_table, aes(x = pgs_midpoint, y = odds_ratio, ymin = ci_lower, ymax = ci_upper)) +
  geom_hline(yintercept = 1.0, color="darkgray", linetype = "dotted") +
  geom_errorbar(width=0.3, color="darkgray") + 
  geom_point(size=4) +
  geom_line(data = predicted_or, 
            aes(x = pgs_quantile_val, y = pred_or, group=1), color = "darkred", inherit.aes = FALSE) +
  geom_segment(data = pgs_thresholds, aes(x=0, xend = pgs_cutoff, y = or_thresh, yend = or_thresh, color = thresh_lab), inherit.aes = F, linetype = "dashed") +
  geom_segment(data = pgs_thresholds, aes(y=0, yend = or_thresh, x =pgs_cutoff, xend=pgs_cutoff, color = thresh_lab), inherit.aes = F, linetype = "dashed") +
  geom_text(data = pgs_thresholds, aes(x=0, y = or_thresh, label = paste0(thresh_lab, " OR=",or_thresh, " (", sprintf("%.0f%%", pgs_cutoff), ")"),  color = thresh_lab), vjust=-1, hjust=0, inherit.aes = F, show.legend = FALSE) +
  theme_classic(base_size = 22) +
  labs(x = "PGS quantile", 
       y = "OR for DCM") +
  guides(color = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=1))
p3
png(file.path(dir, "figures", "odds_ratio_psg.png"), width=600, height=600)
p3
invisible(dev.off())

# plot the distributions of PGS in affected and unaffected
p2 <- ggplot(dat, aes(x = pgs, fill = dcm_status)) +
  geom_density(alpha=0.3) +
  geom_vline(data = pgs_thresholds, aes(xintercept = pgs), color="red") +
  geom_text(data = pgs_thresholds, aes(x=pgs, y = 0, label = paste0(thresh_lab, " OR=",or_thresh)), size=6, color = "red", vjust=-0.5, hjust=0, angle=90, inherit.aes = F, show.legend = FALSE) +
  scale_fill_viridis_d() +
  labs(x = "PGS", fill = "DCM status") +
  theme_classic(base_size = 22) +
  theme(axis.text.x  = element_blank(), 
        axis.ticks.x = element_blank(), 
        axis.text.y  = element_blank(), 
        axis.ticks.y = element_blank(), 
        axis.title.y = element_blank())
p2
png(file.path(dir, "figures", "psg_dcm_distributions.png"), width=800, height=600)
p2
invisible(dev.off())
```

### Odds ratio thresholding and increase in diagnostic yeild
We then annotate the DCM cases in the dataset as to whether they have a pathogenic, intermediate or low PGS. 

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.align='center', out.width="50%"}
# plot the pie chart of DCM gene status by PGS thresholds
sunburst_data <- rbind(
  dat[dcm_status == "DCM", .(gene_status = "DCM", thresh_lab = "", N = .N)],
  dat[dcm_status == "DCM", .(thresh_lab = "", N = .N), by = "gene_status"],
  dat[dcm_status == "DCM", .N, by = .(gene_status, thresh_lab)]
)[, `:=`(id     = trimws(paste(gene_status, thresh_lab)),
         parent = fcase(gene_status=="DCM", "", 
                        thresh_lab==""    , "DCM", 
                        thresh_lab!=""    , as.character(gene_status)))]
sunburst_data[, label := fcase(gene_status=="DCM", "DCM", 
                               thresh_lab==""    , as.character(gene_status),
                               thresh_lab!=""    , thresh_lab)]
sunburst_data <- sunburst_data[, .(id, parent, N, label)]

# Create the sunburst plot
p4 <- plot_ly(
  sunburst_data,
  ids     = ~id,
  labels  = ~label,
  parents = ~parent,
  values  = ~N,
  type    = 'sunburst',
  text = ~N,
  branchvalues = 'total'
) |> layout(colorway = c("#345799", "#e6550d", "#9ecae1"))
p4
save_image(p4, file.path(dir, "figures", "gene_status_by_psg_threshold.png"))
```


## Likelihood ratios and population attributable risk
We derive the odds-ratio per standard deviation chnage in PGS for DCM use this to calculate the likelihood ratios, and population attributable risk, by PGS bin.

```{r}
# Likelihood ratios and attributable risk faction
# calculate the odds ratio per Sdev of the PGS
model             <- glm(dcm_status ~ pgs, data = dat, family = "binomial")
summary_model     <- summary(model)
odds_ratio_sd     <- exp(coef(summary_model)["pgs", "Estimate"] * sd(dat$pgs))
odds_ratio_sd_lci <- exp((coef(summary_model)["pgs", "Estimate"] - 1.96*coef(summary_model)["pgs", "Std. Error"]) * sd(dat$pgs))
odds_ratio_sd_uci <- exp((coef(summary_model)["pgs", "Estimate"] + 1.96*coef(summary_model)["pgs", "Std. Error"]) * sd(dat$pgs))

attributable_risk_frac <- function(prev, or_per_sd, centile) {
  odds   <- prev / (1-prev)
  u.A    <- log(or_per_sd)
  u.U    <- 0
  u.diff <- u.A - u.U
  z.U    <- qnorm(centile)
  z.A    <- z.U - u.A
  pgs_centile_aff <- pnorm(z.A)
  L.U    <- dnorm(z.U)
  L.A    <- dnorm(z.A)
  LR     <- L.A / L.U
  posttest.odds <- odds * LR
  posttest.P    <- posttest.odds / (posttest.odds + 1)
  afe           <- 1 - (prev / posttest.P)
  list(LR = LR, afe = afe)
}

likelihood_ratio <- data.table(
  dcm.P             = dcm_prev, 
  pgs_or_per_sd     = odds_ratio_sd,
  pgs_or_per_sd_lci = odds_ratio_sd_lci,
  pgs_or_per_sd_uci = odds_ratio_sd_uci,
  pgs_centile_unaff = seq(0.01,0.99,0.01)
)
likelihood_ratio[, c("LR", "afe")         := attributable_risk_frac(dcm.P, pgs_or_per_sd, pgs_centile_unaff)]
likelihood_ratio[, c("LR_lci", "afe_lci") := attributable_risk_frac(dcm.P, pgs_or_per_sd_lci, pgs_centile_unaff)]
likelihood_ratio[, c("LR_uci", "afe_uci") := attributable_risk_frac(dcm.P, pgs_or_per_sd_uci, pgs_centile_unaff)]
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.align='center', out.width="50%"}
# Create the PGS centile vs proportion of DCM cases plot 
sd_lines <- data.table(y = c(ecdf(dat$pgs)(mean(dat$pgs) - 2*sd(dat$pgs)),
                             ecdf(dat$pgs)(mean(dat$pgs) - sd(dat$pgs)),
                             ecdf(dat$pgs)(mean(dat$pgs)),
                             ecdf(dat$pgs)(mean(dat$pgs) + sd(dat$pgs)),
                             ecdf(dat$pgs)(mean(dat$pgs) + 2*sd(dat$pgs))), 
                       x = 1,
                       text  = c("-2SD", "-1SD", "Mean", "+1SD", "+2SD"))

p5 <- ggplot((dat[, .(dcm_count = sum(dcm_status == "DCM"), total_count = .N), by = pgs_percentile]
           [order(as.numeric(pgs_percentile)), ]
           [, `:=`(cumulative_dcm_count   = cumsum(dcm_count))]
           [, `:=`(cumulative_proportion  = cumulative_dcm_count / dat[dcm_status == "DCM", .N])]
           [, `:=`(se                     = sqrt((cumulative_proportion * (1 - cumulative_proportion)) / dat[dcm_status == "DCM", .N]))]), 
       aes(y = as.numeric(pgs_percentile)/100, x = cumulative_proportion, group=1)) +
  geom_hline(data = sd_lines, aes(yintercept=y), linetype = "dotted", linewidth=1) +
  geom_text(data = sd_lines, aes(y=y, x=x, label=text), vjust=-0.3, hjust=0, size=5) +
  geom_line() + 
  geom_ribbon(aes(xmin = cumulative_proportion - 1.96*se, xmax = cumulative_proportion + 1.96*se), alpha = 0.2) + 
  labs(y = "PGS Percentile", x = "Proportion of DCM Cases") +
  scale_y_continuous(breaks = seq(0, 1, by = 0.1), labels = percent) + 
  scale_x_continuous(breaks = seq(0, 1, by = 0.1), labels = percent, expand = c(0,0.1)) + 
  theme_minimal(base_size = 18) +
  theme(panel.grid.minor = element_blank())
p5
png(file.path(dir, "figures", "pgs_by_pct_dcm_cases.png"), width=600, height=500)
p5
invisible(dev.off())


# plot the likelihood ratio by PGS centile
p6 <- ggplot(likelihood_ratio, aes(x = pgs_centile_unaff, y = LR)) +
  geom_line(color = "darkgray") +
  geom_point(size=3) +
  scale_x_continuous(breaks = seq(0,1,0.1), labels = percent) +
  labs(x = "PGS centile in the unaffected", y = "Likelihood Ratio") +
  theme_classic(base_size = 18)
p6
png(file.path(dir, "figures", "lr_by_pgs.png"), width=600, height=500)
p6
invisible(dev.off())

# plot the attributable risk factor for any given PGS centile
p7 <- ggplot(likelihood_ratio[afe>=0], aes(x = pgs_centile_unaff, y = afe)) +
  geom_ribbon(aes(ymin=afe_lci, ymax=afe_uci), alpha = 0.3) +
  geom_point(size=3) +
  scale_y_continuous(breaks = seq(0,1,0.1), labels = percent) +
  scale_x_continuous(breaks = seq(0,1,0.05), labels = percent) +
  labs(x = "PGS centile", y = "Attributable fraction among DCM") +
  theme_classic(base_size = 18)
p7
png(file.path(dir, "figures", "afe_by_pgs.png"), width=600, height=500)
p7
invisible(dev.off())
```



## Introduction

Dilated cardiomyopathy (DCM) is a primary cardiac muscle disorder characterised by abnormal structure and function of the heart with a prevalence of 0.4%. In terms of aetiology, it is an apparently heterogeneous condition with monogenic genetic causes to environmental exposures. DCM is an important diagnosis, carrying high morbidity and mortality rates from complications such as heart failure and arrhythmia. Current genetic testing identifies a culprit gene in approximately 15-25% of sporadic cases and 20-40% of patients with familial DCM. However, the heritability of DCM is XX estimated from common variants suggesting that there exists an underlying propensity to the development of DCM, how this interacts with different exposures over the life course is not fully understood. 

In the clinic we are often faced with so called gene-elusive DCM, patients who display the phenotype but lack a known pathogenic variant. Polygenic risk scores (PGS) can be used to determine the overall burden of DCM associated variants, identified from large genome-wide association studies. PGSs represents the weighted sum on independent genetic variants associated with a disease, in any given individual. There is great interest in how PGSs might be incorporated into clinical use and population screening. [@torkamaniPersonalClinicalUtility2018; @wandImprovingReportingStandards2021] 

PGSs are often interpreted as the amount of risk increase in disease per standard deviation increase in polygenic risk. Conversion of PGSs to clinically meaningful performance metrics (odds of being affected, or positive predictive value) can be derived from the overlap in polygenic risk score distributions between affected and unaffected groups. It is mathematically simple and has been described in detail elsewhere. [@waldAssessingRiskFactors2011; @hingoraniPerformancePolygenicRisk2023a] Additionally, with knowledge of the baseline prevalence of DCM in the general population this can be leveraged to estimate the population attributable risk for an individual.

Currently, it is unclear how the family of gene-elusive patients should be screened and followed up. Current guidelines suggest ... however the penetrance of cases is low and better stratification is needed in order to utilised resources most effectively.

Here, we show how a polygenic risk score for DCM, with appropriate thresholds, can be used in the diagnosis of 'polygenic DCM' and the population attributable risk for any particular genome in order to better communicate to patients the likely aetiology of their underlying condition. 

---

## Methods

#### Study population and phenotypes
Sequencing data for 2,XXX individuals of European ancestry referred for dilated cardiomyopathy were collection from XX centers: X, Y, X. XX European ancestry controls were identified from the UK Biobank.

Rare variant status was determined by ... and individuals codes as: gene positive (pathogenic/like-pathogenic), variant of unknown significance, and gene negative. 

All research participants provided written informed consent, and the studies were reviewed and approved by the relevant research ethics committee (X, Y, Z).

Base data for DCM association statistics were obtained from the latest HERMES consortium GWAS.[@zhengGenomewideAssociationAnalysis2024]

#### Genotyping, quality control and variant annotation
Genotyping as performed using the XX Array. Pre-imputation quality control included X, Y, Z. Imputation was conducted using the Michigan Imputation Server with TOPMed Freeze5 reference panel[???].

#### Polygenic risk score calculation
Common genetic variant effect sizes obtained from previously published GWAS summary statistics were used as weightings in the generation of the PGS for DCM. DCM GWAS summary statistics were filtered to exclude rare variants (MAF < 1%) and ambiguous variants not resolvabnle by strand flipping. 

The XXX software was used for generation of the PGS with parameters X, Y, Z. PGS value for each individual was calculated in the standard way, using genotype dosage for each allele, multiplied by its weight, summed across all variants. [@choiTutorialGuidePerforming2020] 

#### Statistical analysis



---

## Results

#### DCM gene status
The distribution of DCM patients by DCM-gene status is presented in Figure 1. XX% of patients carried and pathogenic or likely-pathogenic genetic variant, XX% of patients carried a variant of unknown significance, and XX% of patients were classified as gene elusive DCM. 

### Polygenic risk score
The distributions of PGS by DCM-gene status and unaffected individuals is presented in Figure 2. The mean PGS was greatest in the gene-positive DCM cohort (XX), followed with the cohort with a variant of unknown significance (XX), and gene negative DCM (XX). Unaffected individuals had the lowest mean PGS (XX). 

The PGS at a 'pathogenic' odds-ratio threshold of >=5 for DCM was XX and for the 'intermediate' odds-ratio threshold of 1.5 was XX (Figure 3). Categorising gene-elusive DCM cases using these thresholds showed that XX% of gene-elusive patients had a pathogenic DCM PGS and XX% had an intermediate PGS (Figure 4).

### Population attributable risk
The population attributable risk (PAR) per PGS centile is presented in Figure 5. Pathogenic and intermediate PGS thresholds had an explained PAR of XX% and XX% respectively.

### Follow up
The survival curves by polygenic risk stratification are shown in Figure X. 

---

## Discussion 
Pathogenic genetic variants explain roughly XX% of cases of dilate cardiomyopathy, leaving a remaining XX% so called gene-elusive cases. The polygenic risk in these remaining cases is substantial and likely explanatory in a significant number of patients.

...

Whilst PGSs can explain significant amounts of variability in traits, concerns have been raised on the variability of different scores, equally performant on the population level, on the individual level, as well as their use in population screening, owing to substantial overlap in PGS distributions amongst the affected and unaffected.[@abramowitzEvaluatingPerformanceAgreement2024; @hingoraniPerformancePolygenicRisk2023a] It has been estimated that an 80% detection rate for a 5% false positive rate requires an odds ratio for one standard deviation increase in PGS of 12. [@hingoraniPerformancePolygenicRisk2023a]

Notwithstanding the limitations of PGSs in defining individual patient risk and population screening, we demonstrate here the utility of PGS in quantifying the underlying polygenic risk in patients who are manifestly DCM patients but in whom a likely single causative genetic variant has not been found. We propose that this is clinically useful information, informing patient discussions regarding their disease and potential implications for family members. 

Strategies to effectively communicate risk estimates derived from PGSs to clinicians and patients are required, especially when communicating results to unaffected individuals where the psychosocial impact could be substantial if not managed appropriately. 

PGSs may help explain the variable penetrance of DCM in family members with identified point mutations but without the clinical phenotype. Polygenic hazard scores could be used to determine the likely age of onset of DCM based on an individual's genetic architecture. [@desikanGeneticAssessmentAgeassociated2017] The genetic architecture of DCM is likely a combination of common low-risk variants (polygenic risk) and rare high-risk (monogenic, familial) variants[@katsanisContinuumCausalityHuman2016]. Incorporating the interaction between polygenic risk and rare variants is desirable but requires further investigation within well designed studies. 


#### Conclusion
Implementation of a dilated cardiomyopathy polygenic risk score can increase the diagnostic yield by X% with 'polygenic' cases of DCM. This has important implications for family screening and ongoing surveillance of individuals with gene-elusive DCM. 

---

## Data and code availability 
The code used to generate these analyses and figures is available on [GitHub](https://github.com/nicksunderland/dcm_pgs). UK Biobank data is available on application. 

## Acknowledgements and funding sources
...

## Supplemental / testing

### Odds ratio by age category testings
```{r, warning=FALSE, message=FALSE, eval=T}
# calculate the odds ratio of DCM by PGS quantile 
dat2 <- copy(dat)
n_quantiles <- 10
quantiles   <- c(0, 1, 5, 10, 20, 40, 60, 80, 90, 95, 99, 100)
dat2[, pgs_quantile := cut(pgs, 
                           breaks = quantile(pgs, probs = quantiles/100, na.rm = TRUE), 
                           labels = c(paste0("<",quantiles[2]), 
                                      paste0(quantiles[2:(length(quantiles)-2)], "-", quantiles[3:(length(quantiles)-1)]), 
                                      paste0(">", quantiles[length(quantiles)-1])), 
                            include.lowest = TRUE), by="age_cat"]
dat2[, pgs_percentile := cut(pgs, 
                            breaks = quantile(pgs, probs = seq(0,1,0.01), na.rm = TRUE), 
                            labels = seq(1,100,1), 
                            include.lowest = TRUE), by="age_cat"]
or_table2 <- expand.grid(pgs_quantile = unique(dat2$pgs_quantile), 
                         age_cat      = unique(dat2$age_cat)) |> as.data.table()

continuity_correction <- 0.5

or_table2[, `:=`(
  dcm_cases_in_group    = dat2[pgs_quantile == .BY[[1]] & age_cat == .BY[[2]] & dcm_status != "No DCM", .N + continuity_correction],
  dcm_controls_in_group = dat2[pgs_quantile == .BY[[1]] & dcm_status == "No DCM", .N + continuity_correction],
  dcm_cases_rest        = dat2[pgs_quantile != .BY[[1]] & age_cat == .BY[[2]] & dcm_status != "No DCM", .N + continuity_correction],
  dcm_controls_rest     = dat2[pgs_quantile != .BY[[1]] & dcm_status == "No DCM", .N + continuity_correction]
), by = .(pgs_quantile, age_cat)]
or_table2[, `:=`(
  odds_in_group = dcm_cases_in_group / dcm_controls_in_group,
  odds_in_rest  = dcm_cases_rest / dcm_controls_rest
)]
or_table2[, `:=`(odds_ratio    = odds_in_group / odds_in_rest, 
                se_log_or     = sqrt(1 / dcm_cases_in_group + 
                                     1 / dcm_controls_in_group + 
                                     1 / dcm_cases_rest + 
                                     1 / dcm_controls_rest))]
or_table2[, `:=`(log_or   = log(odds_ratio))]
or_table2[, `:=`(ci_lower = exp(log_or - 1.96 * se_log_or),
                ci_upper = exp(log_or + 1.96 * se_log_or))]

# model the odds ratio across PGS quantile and extract the thresholds of 5 and 1.5
or_table2[, pgs_midpoint := as.numeric(sub("-.*", "", as.character(pgs_quantile))) + (as.numeric(sub(".*-", "", as.character(pgs_quantile))) - as.numeric(sub("-.*", "", as.character(pgs_quantile)))) / 2]
or_table2[as.character(pgs_quantile) == "<1", pgs_midpoint := 0.5]
or_table2[as.character(pgs_quantile) == ">99", pgs_midpoint := 99.5]


spline_models <- or_table2[, .(
  model = list(lm(log(odds_ratio) ~ ns(pgs_midpoint, df = 6), data = .SD))
), by = age_cat]

pred_pgs_values <- seq(0,100,0.01)
predicted_or <- spline_models[, .(
  pgs_quantile_val = pred_pgs_values,
  pred_or = exp(predict(model[[1]], newdata = data.frame(pgs_midpoint = pred_pgs_values)))
), by = age_cat]

# predict the thresholds from the fit 

pgs_thresholds <- data.table(
  or_thresh  = rep(c(1.5, 5.0), length(unique(predicted_or$age_cat))),
  thresh_lab = rep(c("intermediate", "pathogenic"), length(unique(predicted_or$age_cat))),
  age_cat    = rep(unique(predicted_or$age_cat), 2)
)

pgs_thresholds[, pgs_cutoff := mapply(function(thresh, age) predicted_or[age_cat==age][which.min(abs(pred_or-thresh)), pgs_quantile_val], or_thresh, age_cat)]



foo <- ggplot(or_table2[age_cat!="0-19"], aes(x = pgs_midpoint, y = odds_ratio)) +
  geom_hline(yintercept = 1.0, color="darkgray", linetype = "dotted") +
  #geom_errorbar(aes(ymin = ci_lower, ymax = ci_upper), width=0.3, color="darkgray") + 
  geom_point(size=4) +
  geom_line(data = predicted_or[age_cat!="0-19"], 
            aes(x = pgs_quantile_val, y = pred_or, group=1), color = "darkred", inherit.aes = FALSE) +
  geom_segment(data = pgs_thresholds[age_cat!="0-19"], aes(x=0, xend = pgs_cutoff, y = or_thresh, yend = or_thresh, color = thresh_lab), inherit.aes = F, linetype = "dashed") +
  geom_segment(data = pgs_thresholds[age_cat!="0-19"], aes(y=0, yend = or_thresh, x =pgs_cutoff, xend=pgs_cutoff, color = thresh_lab), inherit.aes = F, linetype = "dashed") +
  geom_text(data = pgs_thresholds[age_cat!="0-19"], aes(x=0, y = or_thresh, label = paste0(thresh_lab, " OR=",or_thresh, " (", sprintf("%.0f%%", pgs_cutoff), ")"),  color = thresh_lab), vjust=-1, hjust=0, inherit.aes = F, show.legend = FALSE) +
  theme_classic(base_size = 22) +
  labs(x = "PGS quantile", 
       y = "OR for DCM") +
  guides(color = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust=1, vjust=1)) +
  facet_wrap(~age_cat, scales="free") 

foo
png(file.path(dir, "figures", "foo.png"), width=1000, height=700)
foo
invisible(dev.off())
```

---

## References