## DCM PGS analysis plan

The initial idea would be to focus on DCM PGS but I’m thinking you could
explore BMI-adjusted and BMI PGS conditional genetic risk in due course.
(linking to the adiposity theme of your PhD). Some sequence experience
would be good to have.

The idea I have is to estimate the proportion of patients with gene
elusive DCM who have disease explained by their DCM PGS. A simple
approach would be partition the PGS (above certain level versus
remainder) and then declare presence or absence of risk. We’re keen to
extend to provide an estimator of population attributable risk per
individual, using likelihood ratios.

The core idea is attached. There are many ways of calculating PAR and
need to kick the tyres on this simplification, but I think this works.

Primary dataset – GoDCM Imperial cohort of 1500 patients +/- another DCM
case cohort

UKB for population stuff

## Paper structure

Possible manuscript structure

GREML h2SNP DCM UKB, versus heritability due to rare variation (will dig
out method) Review RR thresholds on risk and use as a framework to
determine thresholds
<https://clinicalgenome.org/working-groups/low-penetrance-risk-allele-working-group/>
Estimate the proportion of cases with polygenic cause (then test in
subsequent) see below working GoDCM Gene positive – P/ LP in DCM gene
Gene negative – other Distribution of PGS in gene+ versus gene –
Identify gene neg with high polygenic score Apply ‘diagnostic’
thresholds according to above Framework for calculating individual PAR
Using attached framework, plot PAR against PGS centiles Conclude ~1/20
patients with DCM polygenic cause Role of diagnosis in testing
