{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import subprocess\n",
    "import hail as hl\n",
    "from pyspark.sql import SparkSession"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "ab6a4de37c31ce9"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "builder = (\n",
    "    SparkSession\n",
    "    .builder\n",
    "    .enableHiveSupport())\n",
    "spark = builder.getOrCreate()\n",
    "hl.init(sc=spark.sparkContext)\n",
    "hl.default_reference(\"GRCh38\")\n",
    "print(\"Hail version:\", hl.__version__)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "42db7411fbfe18f"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "!hdfs dfs -mkdir -p /tmp/pca_bgen\n",
    "!hdfs dfs -put -f /mnt/project/dcm_pgs/pca_variants/dcm_pca_chr*_subset.bgen /tmp/pca_bgen/\n",
    "!hdfs dfs -ls /tmp/pca_bgen | head -n 10"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "79143ea341816880"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "contig_recoding = {f\"{i:02d}\": str(i) for i in range(1, 23)}\n",
    "\n",
    "for c in range(1, 23):\n",
    "    bgen = f\"hdfs://master:9000/tmp/pca_bgen/dcm_pca_chr{c}_subset.bgen\"\n",
    "    idx2 = bgen + \".idx2\"\n",
    "\n",
    "    print(f\"chr{c}: indexing\")\n",
    "    hl.index_bgen(\n",
    "        path=bgen,\n",
    "        index_file_map={bgen: idx2},\n",
    "        reference_genome=\"GRCh37\",\n",
    "        contig_recoding=contig_recoding,\n",
    "    )\n",
    "\n",
    "print(\"Done indexing.\")"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "c3074be2140495e7"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# BGEN files\n",
    "bgen_paths = [\n",
    "    f\"hdfs://master:9000/tmp/pca_bgen/dcm_pca_chr{c}_subset.bgen\"\n",
    "    for c in range(1, 23)\n",
    "]\n",
    "\n",
    "mt = hl.import_bgen(\n",
    "    bgen_paths,\n",
    "    sample_file=\"file:///mnt/project/Bulk/Imputation/UKB imputation from genotype/ukb22828_c1_b0_v3.sample\",\n",
    "    entry_fields=[\"dosage\"]\n",
    ")"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "72e4e02ecded1bb7"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Annotate rows with loadings and filter to intersection\n",
    "ld = hl.read_table(\"file:///mnt/project/dcm_pgs/loadings/gnomad.v3.1.pca_loadings_grch37.ht\")\n",
    "ld = ld.key_by(\"locus\", \"alleles\")\n",
    "mt = mt.annotate_rows(l = ld[mt.row_key])\n",
    "mt = mt.filter_rows(hl.is_defined(mt.l))"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "734b52758476be61"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "p = mt.l.pca_af\n",
    "mu = 2.0 * p\n",
    "sigma = hl.sqrt(2.0 * p * (1.0 - p))\n",
    "x = (hl.float64(mt.dosage) - mu) / sigma\n",
    "\n",
    "# compute PC1..PC10 without building a vector per entry\n",
    "mt = mt.annotate_entries(\n",
    "    c1 = x * mt.l.loadings[0],\n",
    "    c2 = x * mt.l.loadings[1],\n",
    "    c3 = x * mt.l.loadings[2],\n",
    "    c4 = x * mt.l.loadings[3],\n",
    "    c5 = x * mt.l.loadings[4],\n",
    "    c6 = x * mt.l.loadings[5],\n",
    "    c7 = x * mt.l.loadings[6],\n",
    "    c8 = x * mt.l.loadings[7],\n",
    "    c9 = x * mt.l.loadings[8],\n",
    "    c10 = x * mt.l.loadings[9],\n",
    ")\n",
    "\n",
    "mt = mt.annotate_cols(\n",
    "    PC1 = hl.agg.sum(mt.c1),\n",
    "    PC2 = hl.agg.sum(mt.c2),\n",
    "    PC3 = hl.agg.sum(mt.c3),\n",
    "    PC4 = hl.agg.sum(mt.c4),\n",
    "    PC5 = hl.agg.sum(mt.c5),\n",
    "    PC6 = hl.agg.sum(mt.c6),\n",
    "    PC7 = hl.agg.sum(mt.c7),\n",
    "    PC8 = hl.agg.sum(mt.c8),\n",
    "    PC9 = hl.agg.sum(mt.c9),\n",
    "    PC10 = hl.agg.sum(mt.c10),\n",
    ")\n",
    "mt = mt.select_entries()"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "837b31eeb18edd04"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "pcs = mt.cols().select(\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\"PC7\",\"PC8\",\"PC9\",\"PC10\")"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "7c83571f61c050c4"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# export sharded (prevents executor OOM)\n",
    "pcs.export(\"ukb_gnomad_projected_pcs.tsv.bgz\", parallel=\"header_per_shard\")\n",
    "\n",
    "# copy to project\n",
    "hl.hadoop_copy(\n",
    "    \"ukb_gnomad_projected_pcs.tsv.bgz\",\n",
    "    \"file:///mnt/project/dcm_pgs/ukb_gnomad_projected_pcs.tsv.bgz\"\n",
    ")"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "bbd980b1d7d59354"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
